processing/src/main/java/org/apache/druid/data/input/impl/FlatTextInputFormat.java

```diff
@@ -139,6 +140,16 @@ public boolean equals(Object o)
            Objects.equals(delimiter, that.delimiter);
   }
 
+  @Override
+  public long getWeightedSize(String path, long size)
+  {
+    CompressionUtils.Format compressionFormat = CompressionUtils.Format.fromFileName(path);
+    if (CompressionUtils.Format.GZ == compressionFormat) {
+      return size * CompressionUtils.COMPRESSED_TEXT_WEIGHT_FACTOR;
```

Comment:
Is COMPRESSED_TEXT_WEIGHT_FACTOR specific to GZIP? 

If so, maybe rename to constant to be specific to GZ, also, do we want to consider other compression formats here and similar places?

Reply:
its not, gzip is just the most common, and one that we have data for. We can add others as need arises.